src/test/java/uk/gov/companieshouse/authcodenotification/controller/AuthCodeNotificationControllerTest.java

```diff
@@ -39,20 +47,36 @@ class AuthCodeNotificationControllerTest {
     void setup() {
         sendEmailRequestDto = new SendEmailRequestDto();
         sendEmailRequestDto.setAuthCode(AUTH_CODE);
+        when(dataSanitiser.makeStringSafe(COMPANY_NUMBER)).thenReturn(COMPANY_NUMBER);
+        when(dataSanitiser.makeStringSafe(AUTH_CODE)).thenReturn(AUTH_CODE);
     }
 
     @Test
     void testSendEmailReturnsSuccess() {
+        Errors errors = new Errors();
+        when(authCodeEmailValidator.validate(eq(COMPANY_NUMBER), eq(AUTH_CODE), any(), eq(REQUEST_ID))).thenReturn(errors);
         ResponseEntity<Object> responseEntity = controller.sendEmail(REQUEST_ID, sendEmailRequestDto, COMPANY_NUMBER);
         assertEquals( 200, responseEntity.getStatusCode().value() );
     }
 
     @Test
     void testSendEmailReturnsInternalServerErrorWhenServiceCallFails() throws ServiceException {
-        when(dataSanitiser.makeStringSafe(COMPANY_NUMBER)).thenReturn(COMPANY_NUMBER);
-        when(dataSanitiser.makeStringSafe(AUTH_CODE)).thenReturn(AUTH_CODE);
+        Errors errors = new Errors();
+        when(authCodeEmailValidator.validate(eq(COMPANY_NUMBER), eq(AUTH_CODE), any(), eq(REQUEST_ID))).thenReturn(errors);
         doThrow(new ServiceException("")).when(authCodeNotificationService).sendAuthCodeEmail(REQUEST_ID, AUTH_CODE, COMPANY_NUMBER);
         ResponseEntity<Object> responseEntity = controller.sendEmail(REQUEST_ID, sendEmailRequestDto, COMPANY_NUMBER);
         assertEquals( 500, responseEntity.getStatusCode().value() );
     }
+
+    @Test
+    void testSendValidationFailureReturnsBadRequest() {
+        Errors errors = new Errors();
+        final String errorLocation = "EXAMPLE_ERROR_LOCATION";
+        final String error = "EXAMPLE_ERROR";
+        Err err = Err.invalidBodyBuilderWithLocation(errorLocation).withError(error).build();
+        errors.addError(err);
+        when(authCodeEmailValidator.validate(eq(COMPANY_NUMBER), eq(AUTH_CODE), any(), eq(REQUEST_ID))).thenReturn(errors);
+        ResponseEntity<Object> responseEntity = controller.sendEmail(REQUEST_ID, sendEmailRequestDto, COMPANY_NUMBER);
+        assertEquals( 400, responseEntity.getStatusCode().value() );
```

Comment:
we should also assert on the response body to make sure it is returning the correct fields and message

Reply:
Test response body returns the error input